system/jhtree/jhtree.cpp

```diff
@@ -811,7 +851,7 @@ class CNodeCache : public CInterface
     {
         CriticalBlock block(lock[type]);
         unsigned oldV = cache[type].setMemLimit(newSize);
-        cacheEnabled[type] = (newSize != 0);
+        cache[type].enabled = (newSize != 0);
```

Comment:
minor: could be inside setMemLimit(), or use the memLimit to determine isEnabled() directly.

Reply:
Moved